### ISSUE TITLE
bignum.h : error: ‘CHAR_BIT’ was not declared in this scope

### ISSUE THREAD

**Issue by nmmmnu** [rupaya-project/rupx#3]
This issue tracker is only for technical issues related to RUPAYA Core.
General RUPAYA questions and/or support requests and are best directed to the [RUPAYA Discord](https://discord.rupx.io).

### Describe the issue

While compiling, following error come up:

libzerocoin/bignum.h: In static member function ‘static CBigNum CBigNum::randBignum(const CBigNum&)’:
libzerocoin/bignum.h:774:55: error: ‘CHAR_BIT’ was not declared in this scope
         size_t size = (mpz_sizeinbase (range.bn, 2) + CHAR_BIT-1) / CHAR_BIT;

#### If so, please list the steps to reproduce below:
1. Compile on Cent OS 7 with GCC 4.8.5

### Expected behavior
Tell us what should happen

pass normally

### Actual behavior
Tell us what happens instead

libzerocoin/bignum.h: In static member function ‘static CBigNum CBigNum::randBignum(const CBigNum&)’:
libzerocoin/bignum.h:774:55: error: ‘CHAR_BIT’ was not declared in this scope
         size_t size = (mpz_sizeinbase (range.bn, 2) + CHAR_BIT-1) / CHAR_BIT;

### Screenshots.
If the issue is related to the GUI, screenshots can be added to this issue via drag & drop.

### What version of RUPAYA Core are you using?
List the version number/commit ID, and if it is an official binary, self compiled or a distribution package.

### Machine specs:
- OS: CentOS 7
- CPU:
- RAM:
- Disk size:
- Disk Type (HD/SDD):

### Any extra information that might be useful in the debugging process.
This is normally the contents of a `debug.log`, `db.log` or `config.log` file. Raw text or a link to a pastebin type site are preferred.

Fix of the problem, 

in libzerocoin/bignum.h, add

#include <climits>

**Comment by binarysculptor** [comment]
Please try to use configure options and see if that fixes your issue.

tldr: `--with-zerocoin-bignum gmp`

detailed: https://github.com/PIVX-Project/PIVX/pull/681

**Comment by nmmmnu** [comment]
i was able to compile. but you need to document this.
